src/rust-crypto/backup.ts

```diff
@@ -124,6 +145,43 @@ export class RustBackupManager extends TypedEventEmitter<RustBackupCryptoEvents,
         }
         return this.keyBackupCheckInProgress;
     }
+
+    /**
+     * Handles a backup secret received event and store it if it matches the current backup version.
+     *
+     * @param secret - The secret as received from a `m.secert.send` event for secret `m.megolm_backup.v1`.
+     * @returns true if the secret is valid and has been stored, false otherwise.
+     */
+    public async handleBackupSecretReceived(secret: string): Promise<boolean> {
+        // Currently we only receive the decryption key without any key backup version, it is important to
+        // check that the secret is valid for the current version before storing it.
+        // We force a check to ensure to have the latest version. We also want to check that the backup is trusted
+        // as we don't want to store the secret if the backup is not trusted, and eventually import megolm keys later from an untrusted backup.
+        const backupCheck = await this.checkKeyBackupAndEnable(true);
+        if (backupCheck?.backupInfo?.version && backupCheck.trustInfo.trusted) {
+            try {
+                const backupDecryptionKey = RustSdkCryptoJs.BackupDecryptionKey.fromBase64(secret);
+                const privateKeyMatches = this.backupInfoMatchesBackupDecryptionKey(
+                    backupCheck.backupInfo,
+                    backupDecryptionKey,
+                );
+                if (!privateKeyMatches) {
+                    logger.debug(`onReceiveSecret: backup decryption key does not match current backup version`);
+                    // just ignore the secret
+                    return false;
+                }
+                logger.info(`onReceiveSecret: Received matching secret ${secret}, store it.`);
```

Comment:
storing the secret key in a rageshake seems unwise.

Generally this seems like a rather cryptic log line. Can you make it easier for a reader to understand what it means? Eg, what sort of secret?

Reply:
good catch, got lost with all the refactoring from review was the secret name initialy.